website/orders/admin.py

```diff
@@ -37,6 +38,16 @@ class ProductAdmin(admin.ModelAdmin):
 
     actions = ["make_available", "make_unavailable"]
 
+    def clean_barcode(self):
+        """Clean barcode."""
+        if self.cleaned_data["barcode"] == "" or self.cleaned_data["barcode"] is None:
+            return None
+
+        if len(self.cleaned_data["barcode"]) == 8 or len(self.cleaned_data["barcode"]) == 13:
+            return self.cleaned_data["barcode"]
+        else:
+            raise ValidationError("A barcode must be either 8 or 13 integers long")
```

Comment:
I think a validator ([https://docs.djangoproject.com/en/3.0/ref/validators/)(https://docs.djangoproject.com/en/3.0/ref/validators/)) would be better as it is a restriction on the model-level and not admin-level

Reply:
Will change.

Reply:
Okay, had to make it a CharField again as we can't get the len(int) of an integer. A char field is also more nice in the admin panel for a barcode.

Reply:
Done, it also validates the barcode checksum now.